### ISSUE TITLE
Provide this as a service (like travis)

### ISSUE THREAD

**Issue by hunkim** [hunkim/translation_coverage#11]
Perhaps, we need something like this:
## Web service
- [ ] Web page to login using github OAuth 
- [ ] List projects from the user and select on/off
  - https://developer.github.com/v3/repos/#list-your-repositories
- [ ] Add webhook automatically using the OAuthkey when a project is on
  - https://developer.github.com/v3/orgs/hooks/#create-a-hook
## Webhook service
- [ ] When there is a PR or push, we clone repos and do our job
  - https://developer.github.com/webhooks/
- [ ] Report the results back to repos
  - Can we push? 
  - Or we maintain the results in our server and have a link using a badge or something

**Comment by mingrammer** [comment]
Do you think any web framework or language for web service?

**Comment by mingrammer** [comment]
I'm willing to contribute about it :)

**Comment by mingrammer** [comment]
Do you think any web framework or language for web service?

**Comment by mingrammer** [comment]
And I wonder whether this service is only for Korean. (except Japanese, Chinese and others?)

I think it can also support Japanese and Chinese right now! (but I don't know about other languages)

**Comment by hunkim** [comment]
I really appreciate your commitment! Perhaps, could you check if I miss something in the todo items?

Surely, it should support any languages. 

For web service, I'm thinking of AngularjS2 + Firebase + github host.
For wekbook service, we may use AWS lambda + S3 using PyGithub.

I'm new to both, but it's always fun to learn something new. Any thoughts?

**Comment by mingrammer** [comment]
@hunkim Hm, ah, Why do you separate the service to two services?

I think this service like this:

We provide a web service (translation_coverage) like travis. And, in web service, we also use the webhook internally.

User > Login > Register Repository in our web service > We are waiting the webhooks from github > Run the `translation_coverage` when webhook is received

**Comment by hunkim** [comment]
I agree. However, we need to develop two different services (web for human, and webhook for the github interactions)?

**Comment by mingrammer** [comment]
Could you explain the whole service architecture to me? (At technically point)

**Comment by mingrammer** [comment]
When does start it?

**Comment by hunkim** [comment]
Should we simply implement it as a web-hook? When there is PR/PUSH, we clone and tun this and push back the results. 

What do you think?

**Comment by mingrammer** [comment]
Yes, we should implement the web-hook first. And will make web service using the hooking service later. (after web-hook is done)

Additionally, I can develop the web service (I am web developer (almost backend))

**Comment by mingrammer** [comment]
Will we make the hooking/web service in here? (This repository)

**Comment by hunkim** [comment]
Yes.

**Comment by mingrammer** [comment]
Could you build a structure of web-hook service here?

**Comment by hunkim** [comment]
I think we can use the heruku + flask something like this, https://github.com/DeepLearningProjects/poem-bot?

**Comment by mingrammer** [comment]
I'm ok. I also know flask too. 
So, we should build a flask + heroku environments and make current translation coverage program to a single module for reusability. right?

**Comment by mingrammer** [comment]
How about it? (Very simple version)

/root
    /web (flask based)
    /hook (web-hook related)
    /coverage (as core util)

**Comment by hunkim** [comment]
+1

**Comment by mingrammer** [comment]
Um, ok.

How can we work about that? Any ideas?
